Otherworldly Pines/Assets/Scripts/Gravity Flip/GravityControl.cs

```diff
@@ -1,18 +1,37 @@
-﻿using System.Collections;
+﻿using System;
+using System.Collections;
 using System.Collections.Generic;
 using UnityEngine;
 
 public class GravityControl : GravityAffected, IHUDConnected {
 
     private GravityFlipIndicator indicator;
     private GravityRegion activeGravityRegion;
+    private PlayerMovement player;
+    private bool canFlip = true;
+
+    private void Awake()
+    {
+        player = GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerMovement>();
+    }
 
     void Update()
     {
-        if (Input.GetKeyDown(KeyCode.W) && activeGravityRegion != null)
+        if (Input.GetKeyDown(KeyCode.W) && isFlipActive())
         {
             activeGravityRegion.FlipGravity();
+            canFlip = false;
         }
+
+        if (player.isPlayerGrounded())
+        {
+            canFlip = true;
+        }
```

Comment:
You can flip twice if you're grounded rn because if the player is grounded when gravity is flipped, this resets canFlip to true. What if we just have isFlipActive check if the player is grounded?

Reply:
I implemented it this way so that the player could jump into a grav flip, otherwise gravity flip would just be a fancy jump

Reply:
I updated it so that the player can no longer do that

Reply:
Oh wait shit that's a good point tho